src/main/java/ch/njol/skript/effects/EffBroadcast.java

```diff
@@ -77,44 +85,74 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 			receivers.addAll(Bukkit.getOnlinePlayers());
 			receivers.add(Bukkit.getConsoleSender());
 		} else {
-			for (World world : worlds.getArray(e))
+			for (World world : worlds.getArray(event))
 				receivers.addAll(world.getPlayers());
 		}
 
 		for (Expression<?> message : getMessages()) {
 			if (message instanceof VariableString) {
-				BaseComponent[] components = BungeeConverter.convert(((VariableString) message).getMessageComponents(e));
+				if (!dispatchEvent(getRawString(event, (VariableString) message), receivers))
+					continue;
+				BaseComponent[] components = BungeeConverter.convert(((VariableString) message).getMessageComponents(event));
 				receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
 			} else if (message instanceof ExprColoured && ((ExprColoured) message).isUnsafeFormat()) { // Manually marked as trusted
-				for (Object realMessage : message.getArray(e)) {
+				for (Object realMessage : message.getArray(event)) {
+					if (!dispatchEvent(Utils.replaceChatStyles((String) realMessage), receivers))
+						continue;
 					BaseComponent[] components = BungeeConverter.convert(ChatMessages.parse((String) realMessage));
 					receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
 				}
 			} else {
-				for (Object messageObject : message.getArray(e)) {
+				for (Object messageObject : message.getArray(event)) {
 					String realMessage = messageObject instanceof String ? (String) messageObject : Classes.toString(messageObject);
+					if (!dispatchEvent(Utils.replaceChatStyles(realMessage), receivers))
+						continue;
 					receivers.forEach(receiver -> receiver.sendMessage(realMessage));
 				}
 			}
 		}
 	}
 
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "broadcast " + messageExpr.toString(event, debug) + (worlds == null ? "" : " to " + worlds.toString(event, debug));
+	}
+
 	private Expression<?>[] getMessages() {
 		if (messageExpr instanceof ExpressionList && !messageExpr.getAnd()) {
 			return new Expression[] {CollectionUtils.getRandom(messages)};
 		}
 		return messages;
 	}
 
-	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return "broadcast " + messageExpr.toString(e, debug) + (worlds == null ? "" : " to " + worlds.toString(e, debug));
+	/**
+	 * @param message the message for {@link BroadcastMessageEvent}
+	 * @return true if the dispatched event does not get cancelled
+	 */
+	@SuppressWarnings({"BooleanMethodIsAlwaysInverted, deprecation"})
+	private static boolean dispatchEvent(String message, List<CommandSender> receivers) {
+		BroadcastMessageEvent broadcastEvent = new BroadcastMessageEvent(false, message, new HashSet<>(receivers));
+		Bukkit.getPluginManager().callEvent(broadcastEvent);
+		return !broadcastEvent.isCancelled();
 	}
-	
+
+	@Nullable
+	private static String getRawString(Event event, Expression<? extends String> string) {
+		if (string instanceof VariableString)
```

Comment:
Probably better for another PR once this one is merged but if we're gonna just take an existing method in another class we might as well add this method to `StringUtils`

Reply:
I would love to honestly, but I'm afraid that such changes seems out of scope of this PR for the team. So I'll leave this as is.